### PR TITLE
Site: update network options for start command

### LINKED ISSUE #14663: Minikube container not using docker network specified in the --network option
### What Happened?

I have a couple of docker containers running locally in the same network on my machine. I want to be able to reach the pods in the cluster started with the same docker network from any of the containers running outside the cluster on my machine. I assume since the cluster was started with the same docker network, this should be somehow possible. 

### Attach the log file

N/A

### Operating System

Ubuntu

### Driver

Docker

### PR DESCRIPTION
**Pull request by paularah**
Fixes #14663

To start minikube with a user-defined bridge network, You need both a subnet and a gateway. This applies to both the docker and podman drivers. See https://docs.podman.io/en/latest/markdown/podman-network-create.1.html. 

This PR includes that in the documentation and adds the missing gateway option from the start command documentation.

### CODE REVIEW

**Comment by paularah** [comment]
Hey @klaases, pls have a look

**Review by spowelljr — CHANGES_REQUESTED** [review]
Hi @paularah, `site/content/en/docs/commands/start.md` is a generated file. If you want to update flag descriptions you'll need to update them in https://github.com/kubernetes/minikube/blob/master/cmd/minikube/cmd/start_flags.go and then run `make generate-docs`. Also, `gateway` is not a supported flag for the start command.

**Comment by paularah** [comment]
@spowelljr  Thanks.  I've  made the neccesary changes. 

If you start minikube with a user-defined brigde network, then you need to explicitly create the network with both a subnet and gateway or minikube will throw  this error `docker: Error response from daemon: user specified IP address is supported only when connecting to networks with user configured subnets.`  for a missing subnet or default back to the minikube network for a missing gateway. 

This update makes that clear in the docs.

File: cmd/minikube/cmd/start_flags.go [hunk 1]
```diff
@@ -192,7 +192,7 @@ func initMinikubeFlags() {
 	startCmd.Flags().Bool(noKubernetes, false, "If set, minikube VM/container will start without starting or configuring Kubernetes. (only works on new clusters)")
 	startCmd.Flags().Bool(deleteOnFailure, false, "If set, delete the current cluster if start fails and try again. Defaults to false.")
 	startCmd.Flags().Bool(forceSystemd, false, "If set, force the container runtime to use systemd as cgroup manager. Defaults to false.")
-	startCmd.Flags().StringP(network, "", "", "network to run minikube with. Now it is used by docker/podman and KVM drivers. If left empty, minikube will create a new network.")
+	startCmd.Flags().StringP(network, "", "", "Network to run minikube with. Now it is used by docker/podman and KVM drivers. If left empty, minikube will create a new network. User-defined docker/podman brigde networks need to be created with both a subnet and gateway.")
```
Comment (spowelljr) [inline cmd/minikube/cmd/start_flags.go:195]:
```suggestion
	startCmd.Flags().StringP(network, "", "", "Network to run minikube with. Now it is used by docker/podman and KVM drivers. If left empty, minikube will create a new network. User-defined docker/podman bridge networks need to be created with both a subnet and gateway.")
```

**Review by spowelljr — CHANGES_REQUESTED** [review]

**Review by spowelljr — CHANGES_REQUESTED** [review]
Can you run `make generate-docs` to update the typo in the documentation

**Comment by paularah** [comment]
@spowelljr I think this is alright now. Let me know if there are any changes I need to make.

**Pull request closed without merging**